lib/chef/provider/osx_profile.rb

```diff
@@ -0,0 +1,255 @@
+#
+# Author:: Nate Walck (<nate.walck@gmail.com>)
+# Copyright:: Copyright (c) 2015 Facebook, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/log'
+require 'chef/provider'
+require 'chef/resource'
+require 'chef/resource/file'
+require 'uuidtools'
+
+class Chef
+  class Provider
+    class OsxProfile < Chef::Provider
+      include Chef::Mixin::Command
+      provides :osx_profile, os: "darwin"
+      provides :osx_config_profile, os: "darwin"
+
+      def whyrun_supported?
+        true
+      end
+
+      def load_current_resource
+        @current_resource = Chef::Resource::OsxProfile.new(@new_resource.name)
+        @current_resource.profile_name(@new_resource.profile_name)
+
+        all_profiles = get_installed_profiles
+        @new_resource.profile(
+          @new_resource.profile ||
+          @new_resource.profile_name
+        )
+
+        @new_profile_hash = get_profile_hash(@new_resource.profile)
+        @new_profile_hash['PayloadUUID'] =
+          config_uuid(@new_profile_hash) if @new_profile_hash
+
+        if @new_profile_hash
+          @new_profile_identifier = @new_profile_hash['PayloadIdentifier']
+        else
+          @new_profile_identifier = @new_resource.identifier ||
+          @new_resource.profile_name
+        end
+
+        current_profile = all_profiles['_computerlevel'].find {
+          |item| item['ProfileIdentifier'] ==
+             @new_profile_identifier
+        }
+        @current_resource.profile(current_profile)
+
+        @current_resource.installed(profile_installed?)
```

Comment:
following up on the other comment in the resource.  this property on the resource needs to be deleted.   and just use `profile_installed?` directly.  you're not even setting the new_resource value so reporting is going to get the wrong values and round tripping state from the provider through the current_resource back to the provider again isn't useful.